Tasks/AzureAppServiceManageV0/azureappservicemanage.ts

```diff
@@ -47,7 +78,23 @@ async function run() {
         tl.debug(`Resource Group: ${resourceGroupName}`);
         var appService: AzureAppService = new AzureAppService(azureEndpoint, resourceGroupName, webAppName, slotName);
         var azureAppServiceUtils: AzureAppServiceUtils = new AzureAppServiceUtils(appService);
+        var configSettings = await appService.get(true);
+        var WebAppKind = webAppKindMap.get(configSettings.kind) ? webAppKindMap.get(configSettings.kind) : configSettings.kind;
+        var isLinuxApp = WebAppKind && WebAppKind.indexOf("linux") !=-1;
+        var isContainerApp = WebAppKind && WebAppKind.indexOf("container") !=-1;
 
+        if((action == "Start Swap With Preview" || action == "Complete Swap" || action == "Cancel Swap") && (isLinuxApp || isContainerApp))
+        {
+            throw Error(tl.loc('SwapWithPreviewNotsupported'));
+        }
+        if(action == "Swap Slots" || action == "Start Swap With Preview" || action == "Complete Swap")
+        {
+            targetSlot = (swapWithProduction) ? "production" : targetSlot;
+            var appServiceSourceSlot: AzureAppService = new AzureAppService(azureEndpoint, resourceGroupName, webAppName, sourceSlot);
```

Comment:
can we move this logic to common function as well

Reply:
The variables are being used at multiple places in code. So if we move it to common function we will have to return the values and re-declare them anyways.